### ISSUE TITLE
Proposal: "Group by" for dashboard cards

### ISSUE THREAD

**Issue by edklimov** [metabase/metabase#5346]
Hello,
First of all thank you for building such a great product!

In my opinion it will be a great thing to support “group by” operations for predefined hierarchies in dashboard cards. I have no idea how painful it is from technical perspective, but to my mind it will certainly have some benefits when single predefined grouping is not enough to explore data:
1. User don’t have to make a transition from dashboard’s view to question’s view and loose sight of other dashboard’s data.
2. There will be no need to place charts with the same series but with different groupings near each other on the same dashboard - it will save much of dashboard’s space.
3. Sometimes it is very useful to encapsulate query building process from users and provide just dashboard interface for interaction.

Just a quick sketch for you to better understand what I mean:

![img6](https://user-images.githubusercontent.com/15221186/27227339-c80693de-52ab-11e7-8414-04cdfdf4a082.png)

I suppose this feature will require a lot of extensive work, so maybe you’ll consider more simple option - to create some kind of “overlay” mechanism for dashboard cards and enable its configuration in dashboard edit mode. In theory, this overlay mechanism can allow to compose multiple dashboard cards in one and show desired card using some controls (in my example it is  bottom button panel). Overlaid cards can be downloaded lazily.
This kind of flexibility will allow to compose collections of dashboard cards in a various and very comfortable for end-user ways (even with semantic grouping).

I’ll very appreciate any feedback on this. Thank you in advance.

**Comment by salsakran** [comment]
Not a bad idea, though we'd probably need to add a grouping template variable for sql templates as a precursor ....

**Comment by santaux** [comment]
Good idea. Also, with redash (https://redash.io/), for example, you can do something like that (but only for tables, not for graphs 😞). It would be really cool feature 👍

**Comment by flamber** [comment]
Related to #3974

**Comment by Etienne-Leveque** [comment]
The feature would be very useful. I have a use case where the client want to be able to choose the granularity of the time series (daily, monthly, quarterly, yearly, ...). Currently I need to create 4 different types of questions.

**Comment by NoamNoogata** [comment]
This feature would be very useful when you are dealing with many metrics that have the same visualizations. e.g. sales amount/units sold/number of orders/profits/etc by date. Instead of creating multiple charts that take a lot of space on the dashboard or combining them in a single line chart (when they are in different units or scale). 
I would also note that in other visualizations platforms such as Tableau, QlikSense, PowerBI it is very common. 
It would be great to add this feature to Metabase.

**Comment by alexandroabalo** [comment]
I would love if we could have this feature available! :)  It's very common to have many metrics grouped by different variables.

**Comment by abecquet77** [comment]
Now that the Time grouping parameter is very effective, a feature for grouping according to a parameter would be very welcome.
The work around slow down the requests considerably :
```
SELECT
  yyyymm,
  CASE {{dimension_split}}
    WHEN 'Segment' THEN segment
    WHEN 'Customer Seniority' THEN customer_seniority
    WHEN 'Country' THEN country_name
    WHEN 'Customer Industry' THEN corporate_industry
  END AS dimension,
  SUM(revenue) AS revenue
FROM table
GROUP BY
  yyyymm,
  dimension
```

**Comment by j-kyri** [comment]
Hi there, any guidance on if and when this very useful and very long requested feature might land?
